src/ifcgeom/IfcGeomRepresentation.h

```diff
@@ -176,13 +176,15 @@ namespace IfcGeom {
 							std::vector<std::pair<int,int> > edges_temp;
 
 							const TColgp_Array1OfPnt& nodes = tri->Nodes();
+                            /// @todo can these UVs be used for texture coordinates as is?
```

Comment:
I think your box projection method is more convenient to end-users. These UVs probably have all different scales depending on their underlying geometric form. At least for curved vs planar surfaces.